pkg/autodiscovery/procdiscovery/discover.go

```diff
@@ -0,0 +1,110 @@
+package procdiscovery
+
+import (
+	"encoding/json"
+	"expvar"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/procmatch"
+)
+
+// DiscoverIntegrations retrieves processes running on the host and tries to find possible integrations
+func DiscoverIntegrations(discoverOnly bool) (DiscoveredIntegrations, error) {
+	di := DiscoveredIntegrations{}
+	matcher, err := procmatch.NewDefault()
+
+	if err != nil {
+		return di, fmt.Errorf("Couldn't build default process matcher: %s", err)
+	}
+
+	processes, err := pollProcesses()
+
+	if err != nil {
+		return di, fmt.Errorf("Couldn't retrieve process list: %s", err)
+	}
+
+	if !discoverOnly {
+		running, failing, err := retrieveIntegrations()
+		if err != nil {
+			return di, err
+		}
+		di.Running = running
+		di.Failing = failing
+	}
+
+	// processList is a set of processes (removes duplicate processes)
```

Comment:
why would get dupes? I would expect to get one process per pid from `pollProcesses`